### ISSUE TITLE
Caching Picasso vs. OkHttp

### ISSUE THREAD

**Issue by deveth0** [square/picasso#1131]
Hi,

I've got a weird behavior in my application:
To download images, i use OkHttp with a cache setup. This works fine, as soon as the image has been loaded once, it is served from cache.
I'd now suspect that if I use OkHttpDownloader with Picasso to get the same image, it should be served from cache too. But this does not happen. The first request goes through and only latter requests are cached responses.
Is there any way to solve this?
This is how I setup my Picasso instance:

    new Picasso.Builder(context).downloader(new OkHttpDownloader(client)).build();
And this is a single call:

    picasso.load(source).get();

Thanks a lot!

**Comment by SureshCS-50** [comment]
Image Caching will work from Picasso 2.5.2
your picasso instance setup is correct.

when you are loading your image..

    picasso.load(" url or file ")
                                .networkPolicy(NetworkPolicy.OFFLINE)
                                .placeholder(R.drawable.stub)\
                                .into(imgProfilePic);

networkPolicy(NetworkPolicy.OFFLINE) - will do the magic :)

**Comment by deveth0** [comment]
Yes but this only works if the image is in cache. If it isn't, I want to
have a network call.

**Comment by SureshCS-50** [comment]
First, This will look at your cache. If it is not available, then it will make network call..
Actually this works for me.. Can you test it once.. and update feedback.. :)

**Comment by deveth0** [comment]
If it calls network, this is contrary to the documentation as far as I
understand...

**Comment by SureshCS-50** [comment]
Yup, are you testing in Lollipop devices?

**Comment by deveth0** [comment]
Yes, 5.1

**Comment by SureshCS-50** [comment]
Oook.. Not, I got the issue..
Picasso is not working in Lollipop devices.. But, it is working fine in Kitkat devices..
Am also running behind this issue.. will let you know once i got a solution..

**Comment by deveth0** [comment]
Oh it works fine for me as long as I don't try to mix calls via picasso and
okhttp. This is the only issue I have.

**Comment by SSeanSS** [comment]
@SureshCS-50 my disk cache not working on Android Lollipop as well... Did you manage to solve it now?
